core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -246,8 +246,11 @@ class RequestChannel(val numProcessors: Int, val queueSize: Int) extends KafkaMe
   def sendResponse(response: RequestChannel.Response) {
     if (isTraceEnabled) {
       val requestHeader = response.request.header
-      trace(s"Sending ${requestHeader.apiKey} response to client ${requestHeader.clientId} of " +
-        s"${response.responseSend.size} bytes.")
+      val size = response.responseSend match {
+        case Some(send) => send.size
+        case None => 0
+      }
+      trace(s"Sending ${requestHeader.apiKey} response[action=${response.responseAction}, size=$size] to client")
```

Comment:
But we are not necessarily sending a response. I was thinking of something like:

```scala
response.responseAction match {
  case SendAction => // like now
  case NoOpAction => // log that no response is being sent
  case CloseConnectionAction => // log that the connection is being closed
```

Does that make sense?

Reply:
Agreed. It's clear to do trace logs for different response actions.